modules/order/src/Adjustment.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\commerce_order;
+use Drupal\commerce_price\Price;
+
+/**
+ * Represents an adjustment.
+ */
+final class Adjustment {
+
+  /**
+   * The adjustment type.
+   *
+   * @var string
+   */
+  protected $type;
+
+  /**
+   * The adjustment label.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The decimal amount.
+   *
+   * @var string
+   */
+  protected $amount;
+
+  /**
+   * The currency code.
+   *
+   * @var string
+   */
+  protected $currencyCode;
+
+  /**
+   * The source identifier of the adjustment.
+   *
+   * @var string
+   */
+  protected $sourceId;
+
+  /**
+   * Constructs a new Adjustment instance.
+   *
+   * @param array $definition
+   *   The definition.
+   */
+  public function __construct(array $definition) {
+    foreach (['type', 'label', 'amount', 'currency_code', 'source_id'] as $required_property) {
+      if (empty($definition[$required_property])) {
+        throw new \InvalidArgumentException(sprintf('Missing required property %s.', $required_property));
+      }
+    }
+
+    $this->type = $definition['type'];
+    $this->label = $definition['label'];
+    $this->amount = $definition['amount'];
+    $this->currencyCode = $definition['currency_code'];
+    $this->sourceId = $definition['source_id'];
+  }
+
+  /**
+   * Gets the adjustment type.
+   *
+   * @return string
+   *   The adjustment type.
+   */
+  public function getType() {
+    return $this->type;
+  }
+
+  /**
+   * Gets the adjustment label.
+   *
+   * @return string
+   *   The adjustment label.
+   */
+  public function getLabel() {
+    return $this->label;
+  }
+
+  /**
+   * Gets the decimal amount.
+   *
+   * @return string
+   *   The decimal amount.
+   */
+  public function getDecimalAmount() {
+    return $this->amount;
+  }
+
+  /**
+   * Gets the currency code.
+   *
+   * @return string
+   *   The currency code.
+   */
+  public function getCurrencyCode() {
+    return $this->currencyCode;
+  }
+
+  /**
+   * Get the source identifier.
+   *
+   * @return string
+   *   The source identifier.
+   */
+  public function getSourceId() {
+    return $this->sourceId;
+  }
+
+  /**
+   * Gets the Price value object for the adjustment.
+   *
+   * @return \Drupal\commerce_price\Price
+   *   The Price value object.
+   */
+  public function toPrice() {
```

Comment:
The calculator class works only with amounts, not price objects. So it doesn't matter here. We could replace 'amount' and 'currency_code' wit a single 'amount' that is a Price object, if that makes more sense? Would make it closer to line items.